### PR TITLE
[DOCS] Add collapsible sections to 7.8 breaking changes

### PR DESCRIPTION
**Pull request by jrodewig**
Adds collapsible sections and new format to the 7.8 breaking changes.

Relates to #53229.

### Goals
* Avoid a "wall of text." Let users quickly scan a list of changes to see what's applicable to them.
* Add progressive disclosure, allowing users to get more details only if wanted.
* Add an optional **How to migrate** section for documenting solutions/workarounds.

### Out of scope
* Changing the topic/grouping labels
* Moving all breaking changes for versions within a major (7.x, 8.x, etc.) onto the same page. 

However, we'll want to investigate these ideas more as part of #53229

Preview: http://elasticsearch_55334.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/7.x/breaking-changes-7.8.html

### CODE REVIEW

**Comment by jrodewig** [comment]
If approved, I can make similar changes to the 8.0 breaking changes as part of another PR.

File: docs/reference/migration/migrate_7_8.asciidoc [hunk 1]
```diff
@@ -5,49 +5,65 @@
...
+====
+*Details* +
+Mapping requests that attempt to change the {ref}/enabled.html[`enabled`]
+mapping parameter for a root mapping will fail and return an error.
+
+Previously, {es} accepted mapping requests that attempted to change the
+`enabled` parameter of the root mapping. Theses changes were not applied, but
+such requests didn't return an error.
+
+*How to migrate* +
```
Comment (lcawl) [inline docs/reference/migration/migrate_7_8.asciidoc:38]: I prefer Kibana's more generic "Impact".  IMO you're not telling someone "how to migrate/upgrade" in this section, you're telling them "what do I have to do about it". 

```suggestion
*Impact* +
```

File: docs/reference/migration/migrate_7_8.asciidoc [hunk 2]
```diff
@@ -5,49 +5,65 @@
...
 [[breaking_78_settings_changes]]
 === Settings changes
 
-[discrete]
 [[deprecate-node-local-storage]]
-==== `node.local_storage` is deprecated
+.The `node.local_storage` setting is deprecated
+[%collapsible]
+====
+The `node.local_storage` setting is deprecated. In {es} 8.0.0, all nodes require
```
Comment (lcawl) [inline docs/reference/migration/migrate_7_8.asciidoc:65]: IMO it looks odd that this change doesn't have the same sections as the others.  I don't know this setting, so I'm left wondering what I need to do about this change. Do I have to take action or is this just informational?

Comment (jrodewig) [reply docs/reference/migration/migrate_7_8.asciidoc:66]: I added a **Details** and **Impact** section with 20db1c358f3.

I was originally considering leaving off **Impact** as this setting was previously undocumented and unlikely to be used. However, adding a discontinue instruction doesn't seem harmful.

I guess in situations where we don't have an actionable advice, we could say something along the lines of "No action needed." under **Impact**.

File: docs/reference/migration/migrate_7_8.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/migration/migrate_7_8.asciidoc]: Changed throughout with 20db1c3. Thanks!

**Review by szabosteve — APPROVED** [review]
LGTM, thanks!

**Review by debadair — APPROVED** [review]
LGTM

**Review by lcawl — APPROVED** [review]
LGTM, thanks!

**Comment by jrodewig** [comment]
Thanks all!

**Pull request merged**